### ISSUE TITLE
Camera-focus crash on React Native 0.50.1

### ISSUE THREAD

**Issue by richardgirges** [lwansbrough/react-native-camera#969]
Hey there react-native-camera team. Thanks for such an amazing package, it's done wonders for us!

We've recently upgraded to React Native 0.50.1 and have started getting an app crash when attempting to focus using react-native-camera: `Unsupported top level event type "topFocusChanged" dispatched`.

Exact repro steps:
1. Open view with camera
2. Tap on camera to focus
3. Crash

Full stack trace:
```
Unhandled JS Exception: Unsupported top level event type "topFocusChanged" dispatched
reactConsoleErrorHandler @ ExceptionsManager.js:73
console.error @ YellowBox.js:71
logToConsole @ RCTLog.js:48
logIfNoNativeHook @ RCTLog.js:31
__callFunction @ MessageQueue.js:302
(anonymous) @ MessageQueue.js:116
__guard @ MessageQueue.js:265
callFunctionReturnFlushedQueue @ MessageQueue.js:115
(anonymous) @ debuggerWorker.js:72
```

![image-2](https://user-images.githubusercontent.com/1226126/32410597-f07df074-c181-11e7-8137-6833df262027.jpg)

Here is our `react-native-camera` component code:
```javascript
      <RNCamera
        ref={c => this.camera = c}
        style={{ flex: 1 }}
        deafultOnFocusComponent={true}
        onFocusChanged={e => console.log('FOCUS')}
        keepAwake />
```

package.json:
```
"react-native": "0.50.1",
"react-native-camera": "^0.10.0"
```

Unclear if this is an issue on the react-native-camera side of things or on the RN side of things. Any help would be greatly appreciated!

**Comment by cesarfigueroa** [comment]
We’re experiencing this issue as well on the exact same versions of `react-native` and `react-native-camera`

**Comment by lahlah84** [comment]
Me too

On 6 Nov 2017 21:52, "Cesar Figueroa" <notifications@github.com> wrote:

We’re experiencing this issue as well on the exact same versions of
react-native and react-native-camera

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/lwansbrough/react-native-camera/issues/969#issuecomment-342299633>,
or mute the thread

**Comment by ptomasroos** [comment]
This is because a breaking change in react-native

**Comment by richardgirges** [comment]
Appears to be fixed on react-native 0.50.3 and react-native-camera 0.12.0.
